Automattic/jetpack
# Add page navigation for the new feedback dashboard

## Proposed changes:

This patch implements the page navigation component for the new feedback dashboard, building on top of #28791.  
As we still haven't decided on the URL structure for the new app, nor do we have routing support, the `<PageNavigation />` component reports the current selection by calling the provided `onSelectPage` function.

The component also has an `expandedRange` attribute which allows to control the range of the 'expanded' bit in the middle. I currently set it to `2` which works for all screen sizes, but we could come back to it later and make it switch depending on the current screen size, to include a little more when possible.

**Note:** This PR includes changes from the other PR mentioned above that hasn't been merged yet. The new commits start from `644f612`.

<img width="492" alt="Screenshot 2023-02-08 at 09 55 43" src="https://user-images.githubusercontent.com/8056203/217483983-f08dd800-c0b9-413b-b9c4-8f59b973c4b8.png">

### Other information:


## Jetpack product discussion



pcsBup-Pl-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






Enable the new feedback dashboard with these filters:

```
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
add_filter( 'jetpack_forms_dashboard_enable', '__return_true' );
```

- Go to `/wp-admin/admin.php?page=jetpack-forms`.
- There should be a page navigation component below the responses list.
- Try clicking through the numbers or the previous/next buttons and make sure it works as expected.

Note the page navigation doesn't affect any content yet as we're not yet connected to the backend.
